package.json

```diff
@@ -4,25 +4,61 @@
   "main": "dist/index.js",
   "scripts": {
+    "preinstall": "npx only-allow yarn",
+    "postinstall": "yarn run env:prepare",
+
     "codegen": "yarn exec subql codegen",
     "build": "./scripts/build.sh",
-    "prepack": "rm -rf dist && yarn run build",
+    "watch:build": "./scripts/build.sh no-lint",
+    "lint": "yarn exec eslint src --ext .ts",
+    "lint:fix": "yarn exec eslint src --ext .ts --fix",
+    "format": "yarn exec prettier --loglevel error --write 'src/*.{js,ts,jsx,tsx}'",
+    "format:ci": "yarn exec pretty-quick --verbose --check --pattern 'src/*.{js,ts,jsx,tsx}' --branch origin/main",
     "test": "yarn run build && subql-node-cosmos test",
-    "vendor:setup": "echo 'installing and building vendor modules' && yarn vendor:cosmjs:install && yarn vendor:cosmjs:build && yarn vendor:subql:install && yarn install && yarn vendor:subql:build",
-    "vendor:cosmjs:install": "cd vendor && yarn workspace cosmjs-monorepo-root install",
-    "vendor:subql:install": "cd vendor && yarn workspace @subql/node-cosmos install",
-    "vendor:cosmjs:build": "cd vendor && yarn workspace cosmjs-monorepo-root run build",
-    "vendor:subql:build": "cd vendor && yarn workspace @subql/node-cosmos run build",
-    "vendor:clean": "echo 'cleaning vendor modules' && cd vendor && find . -name node_modules | xargs rm -rf && find . -name dist | xargs rm -rf",
-    "docker:build": "env sh ./scripts/dotenv-check.sh && docker compose build",
-    "docker:build:no-cache": "env sh ./scripts/dotenv-check.sh && docker compose build --no-cache",
-    "docker:start": "docker compose pull && docker compose  up --remove-orphans",
-    "docker:stop": "docker compose stop",
-    "docker:clean": "docker compose down -v",
-    "docker:tunnel": "./scripts/proxy-tunnel.sh"
+    "test:ci": "env node vendor/subql-cosmos/packages/node/bin/run test",
+
+    "vendor:setup": "./scripts/vendor.sh setup",
+    "vendor:clean-cache": "./scripts/vendor.sh clean-cache",
+    "vendor:install": "./scripts/vendor.sh install",
+    "vendor:build": "./scripts/vendor.sh build",
+    "vendor:lint": "./scripts/vendor.sh lint",
+    "vendor:clean": "./scripts/vendor.sh clean",
+
+    "env:prepare": "./scripts/prepare-dotenv.sh",
+    "docker:compose": "./scripts/docker-compose.sh",
+
+    "docker:check-env:production": "env sh ./scripts/dotenv-check.sh production",
+    "docker:build:production": "yarn run docker:compose production build",
+    "docker:build:no-cache:production": "yarn run docker:compose production build --no-cache",
+    "docker:start:production": "yarn run docker:compose production up --remove-orphans",
+    "docker:ps:production": "yarn run docker:compose production ps",
+    "docker:stop:production": "yarn run docker:compose production stop",
+    "docker:clean:production": "yarn run docker:compose production down -v",
+    "docker:tunnel:production": "yarn run docker:check-env:production && ./scripts/proxy-tunnel.sh production",
```

Comment:
Can we reduce `docker:tunnel` to a single yarn script?

Reply:
thing is how it will do the proper checks using the right docker-compose file and dotenv without noticing the environment he is trying to reach.

Reply:
@bryanchriswhite I get it done on the latest commit, please review it again and let me know.